### DISCUSSION TITLE
CMYK to sRGB gives washed out image

### DISCUSSION THREAD

**Discussion by LennardF1989** [dlemstra/Magick.NET#687] [Help] [answered]
I have a TIF file in CMYK (CoatedFOGRA39) and wish to convert that to sRGB (IEC61966-2).

When I do this using the commandline: `convert ZDCF0757.tif -profile CoatedFOGRA39.icc -profile "sRGB Color Space Profile.icm" -strip test.png` I get the output I expect.

However, when I try to replicate this inside Magick.NET, it comes out washed out.

```
using var magic = new MagickImage(input);

magic.Format = MagickFormat.Png;

magic.SetProfile(ColorProfile.CoatedFOGRA39);
magic.SetProfile(ColorProfile.SRGB); //NOTE: Also tried manyally loading the ICC file to be absolutely sure

magic.Strip();

magic.Write(outputStream);
```

I have been toying with the settings for over 2 hours now and it's driving me nuts :P Who can point out what I'm missing? 

Attached is the original TIF and the output of both ImageMagick and Magick.NET with the code above.

[output.zip](https://github.com/dlemstra/Magick.NET/files/4860303/output.zip)

**Comment by dlemstra** [comment]
It looks like you are using an older version of Magick.NET? I cannot reproduce your issue with the latest version. You might also want to change your code to use `magic.TransformColorSpace(ColorProfile.CoatedFOGRA39, ColorProfile.SRGB);` incase another input image has a color profile.

**Comment by LennardF1989** [comment]
I've tried to use the `TransformColorSpace`, but it gave the same result. The attached files have been generated with the latest NuGet package (Q8, but also tried Q16 - AnyCPU, 7.20.1) - which specific version are you using? @dlemstra

EDIT: Additionally: I'm using Windows.

**Comment by dlemstra** [comment]
I was using the latest development build and I think that it still uses the same version of ImageMagick as when I published the new release. Will check it tonight.

**Comment by dlemstra** [comment, marked as answer]
I am unable to reproduce your issue. I tried the following: 

```
<PackageReference Include="Magick.NET-Q8-x64" Version="7.20.0.1" />
```

```C#
using var magic = new MagickImage("ZDCF0757.tif");

magic.Format = MagickFormat.Png;

magic.TransformColorSpace(ColorProfile.CoatedFOGRA39, ColorProfile.SRGB);

magic.Strip();

magic.Write("test.png");
```

And I get the following output:

![test](https://user-images.githubusercontent.com/10426229/86400157-6a2bbb80-bca8-11ea-8366-bb9e1facbe28.png)

**Comment by LennardF1989** [comment]
Very peculiar. I had the issue constantly yesterday, as is evident by the output in the ZIP. But now, it just works... Thanks for checking and confirming!
